packages/template-retail-react-app/app/hooks/use-pickup-shipment.js

```diff
@@ -0,0 +1,184 @@
+     * @returns {string|null} The default shipping method ID, or null if not found
+     */
+    const getDefaultShippingMethodId = (shippingMethods) => {
+        return shippingMethods?.defaultShippingMethodId || null
+    }
+
+    /**
+     * Checks if the current shipping method is already a pickup method
+     * @param {Object} currentShippingMethod - The current shipping method on the basket
+     * @returns {boolean} True if the current shipping method is a pickup method
+     */
+    const isCurrentShippingMethodPickup = (currentShippingMethod) => {
+        return currentShippingMethod?.c_storePickupEnabled === true
+    }
+
+    /**
+     * Ensures pickup shipment is properly configured for the basket
+     * @param {string} basketId - The basket ID
+     * @param {Array} productItems - Array of product items being added
+     * @param {Object} storeInfo - Store information object containing id and inventoryId
+     * @param {Object} options - Configuration options
+     * @param {string} options.pickupShippingMethodId - Shipping method ID for pickup (default: '005')
+     * @param {boolean} options.throwOnError - Whether to throw on error (default: false)
+     */
+    const configurePickupShipment = async (basketId, productItems, storeInfo, options = {}) => {
+        const {pickupShippingMethodId = '005', throwOnError = false} = options
+
+        try {
+            const pickupItems = productItems.filter((item) => item.inventoryId)
+            if (pickupItems.length === 0) return
+
+            if (!storeInfo) {
+                if (throwOnError) throw new Error('Failed to retrieve store information')
+                return
+            }
+
+            if (!storeInfo?.inventoryId) {
+                if (throwOnError) throw new Error('No store inventory ID found')
+                return
+            }
+
+            // Update shipment to ensure pickup configuration
+            await updateShipmentForBasketMutation.mutateAsync({
+                parameters: {
+                    basketId,
+                    shipmentId: 'me'
+                },
+                body: {
+                    shippingMethod: {
+                        id: pickupShippingMethodId
+                    },
+                    c_fromStoreId: storeInfo.id
+                }
+            })
+        } catch (error) {
+            if (throwOnError) {
+                throw error
+            } else {
+                // Log error but don't block the add to cart flow
+                console.warn('Failed to configure pickup shipment:', error)
+            }
+        }
+    }
+
+    /**
+     * Configures regular shipping method for the basket
+     * @param {string} basketId - The basket ID
+     * @param {string} shippingMethodId - The shipping method ID to set
+     * @param {boolean} throwOnError - Whether to throw on error (default: false)
+     */
+    const configureRegularShippingMethod = async (
```

Comment:
`configure` does not indicate api call to me. 
```suggestion
    const updateRegularShippingMethod = async ()
```

Reply:
done